cpp/src/io/parquet/reader_impl_helpers.cpp

```diff
@@ -116,6 +119,10 @@ type_id to_type_id(SchemaElement const& schema,
           return type_id::DECIMAL128;
         }
       }
+      if (physical == parquet::BYTE_ARRAY) {
+        CUDF_EXPECTS(decimal_precision < max_decimal128_precision, "Invalid decimal precision");
```

Comment:
Ah ok, we're asserting that `__int128_t` is big enough here.

Reply:
Is it the case that we always need to send back a `DECIMAL128` here, or could the same handling as for fixed length byte arrays specialise to a narrower type here too?

Reply:
Should this not be `<=`? We want to support a precision of 38 when reading byte arrays.

Reply:
> could the same handling as for fixed length byte arrays specialise to a narrower type here too?

The narrower type would be nice, as the type returned would only depend on the precision in the file and not the physical type it was stored as. But for me it is not critical.

Reply:
"Narrower type" supported